### ISSUE TITLE
x64 build results in -> DLLNotFoundException: Unable to Load DLL 'Microsoft.MixedReality.SceneUnderstanding.dll': The specified module could not be found.

### ISSUE THREAD

**Issue by porter557** [microsoft/MixedReality-SceneUnderstanding-Samples#34]
This is NOT the same as this issue: https://github.com/microsoft/MixedReality-SceneUnderstanding-Samples/issues/28j which has a very similar title. 

I am able to build and run my project just fine in the Unity Editor (2021.3.6f1).

However, our project requires the use of the SceneUnderstanding module to run in an 'AppRemoting' x64 build on a desktop.

These are my build settings in Unity (2021.3.6f1)
![image](https://user-images.githubusercontent.com/119343453/204331097-6a6ee967-03b7-4bd8-883a-804515cda906.png)

Build settings in Visual Studio (VS Community 2019 Version 16.11.19)
![image](https://user-images.githubusercontent.com/119343453/204330545-ff1afd6b-1bb1-49ec-b4d7-ba29018b21bb.png)

The Visual Studio build works just fine! No errors, however, when I attempt to run it. I get the following error:

![image](https://user-images.githubusercontent.com/119343453/204331308-9e09fc35-ff84-4ff4-9673-861f3a77b825.png)

I expected it to be compatible with a desktop build if it could build in Unity and run just fine?

Here's what I have done to attempt to fix it

copied the file '\myProject\Library\PackageCache\com.microsoft.mixedreality.sceneunderstanding@7bfd3a8ea18c\Editor\Plugins\x64\Microsoft.MixedReality.SceneUnderstanding.dll' 

To:
\myProject\build\build\bin\x64\Release\AppX\

When I do this, the SceneUnderstanding.dll is found, but then I get the following error:
![image](https://user-images.githubusercontent.com/119343453/204333000-83baf831-9cb4-4fca-a077-6f9de629998c.png)

I will also add that the documentation says its compatible with UWP build: https://learn.microsoft.com/en-us/windows/mixed-reality/develop/unity/scene-understanding-sdk#where-do-i-get-the-sdk

Any help is greatly appreciated :D Is there a way around this?

**Comment by SzymonSPS** [comment]
Interesting, this is not a case we've seen/tested before. I'll try to repro, but you may be able to test faster than I can given you have this set up. There is a non-0 chance the latest official SDK would handle this case:

https://www.nuget.org/packages/Microsoft.MixedReality.SceneUnderstanding/1.0.14

we've not updated the sample to support it, and some surgery would need to be done. While it's a new major version, the SDK itself should work the same as the previous one, but getting it integrated would mean you need to edit the packages here:

https://github.com/microsoft/MixedReality-SceneUnderstanding-Samples/tree/main/Packages/MixedReality

the new SDK would need to be downloaded and then converted to .tgz so it's rather involved. A simpler hack is to download the nuget, rename to zip, and copy the dll's out and copy them manually to test. You will most likely need to change the code and re-compile, but there's a non-0 chance that it might even work without it, so maybe even repeating your last step: 

"copied the file '\myProject\Library\PackageCache\com.microsoft.mixedreality.sceneunderstanding@7bfd3a8ea18c\Editor\Plugins\x64\Microsoft.MixedReality.SceneUnderstanding.dll'

To:
\myProject\build\build\bin\x64\Release\AppX\" 

using those DLLs work. I'll keep an eye out on this chat to see if I can help.

**Comment by porter557** [comment]
Thank you very much for your response!

I was able to download the new SDK, rename the .nuget to .zip and was able to extract the .dll and copy it into my project in the 
"\myProject\build\build\bin\x64\Release\AppX". I now get this error:

![image](https://user-images.githubusercontent.com/119343453/204846811-04e650d0-1b0b-4668-972d-e5c590010d64.png)

At the very least its not saying that the .dll isn't compatible with a win32 app, so thats something :D

**Comment by SzymonSPS** [comment]
Yes, I think you need to actually upgrade the sample itself to use the newer SDK vs. a drop in. I had a feeling you would hit some breaking changes. But there is a high likelihood of success given the changes from 0.5 to 1.0. We can add upgrading to 1.0 to the backlog but it might take some time just given our schedule.
